opm/simulators/wells/BlackoilWellModelGeneric.cpp

```diff
@@ -1129,25 +1129,17 @@ assignDynamicWellStatus(data::Wells& wsrpt,
                         const int reportStepIndex) const
 {
     this->loopOwnedWells([this, reportStepIndex, &wsrpt]
-                         (const auto wellID, const Well& well)
+                         ([[maybe_unused]] const auto wellID,
+                          const Well& well)
     {
-        auto& xwel = wsrpt[well.name()]; // data::Wells is a std::map<>
 
-        xwel.dynamicStatus = this->schedule()[reportStepIndex]
-            .wells(well.name()).getStatus();
+        auto& xwel = wsrpt[well.name()]; // data::Wells is a std::map<>
+        xwel.dynamicStatus = this->wellState().well(well.name()).status;
 
-        if ((xwel.dynamicStatus == Well::Status::OPEN) &&
-            this->wellTestState().well_is_closed(well.name()) &&
-            !this->wasDynamicallyShutThisTimeStep(wellID))
-        {
-            // Well is supposed to be flowing according to the run
-            // specification (Schedule object), but it is not operable or
-            // cannot meet its economic limits (well testing).
-            //
-            // Assign status based on the well's defined automatic shut-in
-            // procedure.
-            xwel.dynamicStatus = well.getAutomaticShutIn()
-                ? Well::Status::SHUT : Well::Status::STOP;
+        // Well was shut this time step we keep it open one last time
+        // for output
+        if (wasDynamicallyShutThisTimeStep(well.name())) {
```

Comment:
Why did you switch to name-based lookup here?  Is there a reason not to continue using the ID-based lookup?